group-actors/src/main/java/org/sunbird/service/impl/SearchActivityServiceImpl.java

```diff
@@ -0,0 +1,75 @@
+package org.sunbird.service.impl;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.jayway.jsonpath.JsonPath;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.sunbird.models.ActivitySearchRequestConfig;
+import org.sunbird.service.SearchActivityService;
+import org.sunbird.util.ActivitySearchRequestGenerator;
+import org.sunbird.util.HttpClientUtil;
+import org.sunbird.util.JsonKey;
+
+public class SearchActivityServiceImpl implements SearchActivityService {
+  Logger logger = LoggerFactory.getLogger(SearchActivityServiceImpl.class);
+
+  private static SearchActivityService searchActivityService = null;
+  private static ObjectMapper objectMapper = new ObjectMapper();
+
+  public static SearchActivityService getInstance() {
+    if (searchActivityService == null) {
+      searchActivityService = new SearchActivityServiceImpl();
+    }
+    return searchActivityService;
+  }
+
+  @Override
+  public Map<String, Map<String, Object>> searchActivity(List<Map<String, Object>> activities) {
+
+    Map<String, Map<String, Object>> activityInfoMap = new HashMap<>();
+
+    List<ActivitySearchRequestConfig> activitySearchRequestConfigs =
+        ActivitySearchRequestGenerator.generateActivitySearchRequest(activities);
+    try {
+      for (ActivitySearchRequestConfig activitySearchRequest : activitySearchRequestConfigs) {
+        String response =
+            HttpClientUtil.post(
```

Comment:
@sknirmalkar89 - we cant assume that all activity APIs will be POST requests only. Some of the activity types may be GET requests.